homeassistant/components/media_player/dlna_dmr.py

```diff
@@ -0,0 +1,542 @@
+    device = DlnaDmrDevice(hass, dlna_device, name)
+
+    _LOGGER.debug("Adding device: %s", device)
+    async_add_devices([device])
+
+
+class UpnpNotifyView(HomeAssistantView):
+    """Callback view for UPnP NOTIFY messages."""
+
+    url = '/api/dlna_dmr.notify'
+    name = 'api:dlna_dmr:notify'
+    requires_auth = False
+
+    def __init__(self, base_url):
+        """Initializer."""
+        self.base_url = base_url
+        self.event_handler = None
+        self._registered_services = {}
+        self._backlog = {}
+
+    def register(self, app, router):
+        """Register the view with a router."""
+        handler = request_handler_factory(self, self.async_notify)
+        router.add_route('notify', UpnpNotifyView.url, handler)
+
+    async def async_notify(self, request):
+        """Callback method for NOTIFY requests."""
+        headers = request.headers
+        body = await request.text()
+        status = await self.event_handler.handle_notify(headers, body)
+        return aiohttp.web.Response(status=status)
+
+    @property
+    def callback_url(self):
+        """Full URL to be called by device/service."""
+        return urllib.parse.urljoin(self.base_url, self.url)
+
+
+def build_requester(hass):
+    """Build a derived instance of UpnpRequester, specific for hass."""
+    from async_upnp_client import UpnpRequester
+
+    class HassUpnpRequester(UpnpRequester):
+        """async_upnp_client.UpnpRequester for home-assistant."""
+
+        def __init__(self, hass_):
+            """Initializer."""
+            self.hass = hass_
+
+        async def async_do_http_request(self,
+                                        method,
+                                        url,
+                                        headers=None,
+                                        body=None,
+                                        body_type='text'):
+            """Do a HTTP request."""
+            # work around an unknown hass or aiohttp error
+            await asyncio.sleep(0.01)
+
+            session = async_get_clientsession(self.hass)
+            with async_timeout.timeout(5, loop=self.hass.loop):
+                response = await session.request(method,
+                                                 url,
+                                                 headers=headers,
+                                                 data=body)
+                if body_type == 'text':
+                    response_body = await response.text()
+                elif body_type == 'raw':
+                    response_body = await response.read()
+                elif body_type == 'ignore':
+                    response_body = None
+
+            return response.status, response.headers, response_body
+
+    return HassUpnpRequester(hass)
+
+
+class DlnaDmrDevice(MediaPlayerDevice):
+    """Representation of a DLNA DMR device."""
+
+    def __init__(self, hass: HomeAssistant, dmr_device, name=None):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.